plugins/huaweicloud/src/main/java/org/eclipse/xpanse/orchestrator/plugin/huaweicloud/HuaweiTerraformResourceHandler.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.orchestrator.plugin.huaweicloud;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.database.common.DeployResourceKind;
+import org.eclipse.xpanse.modules.database.service.DeployResourceEntity;
+import org.eclipse.xpanse.modules.deployment.DeployResourceHandler;
+import org.eclipse.xpanse.modules.deployment.DeployResult;
+import org.eclipse.xpanse.modules.terraform.deployment.exceptions.TerraformExecutorException;
+import org.eclipse.xpanse.modules.terraform.deployment.resource.TfState;
+import org.eclipse.xpanse.modules.terraform.deployment.resource.TfStateResource;
+import org.eclipse.xpanse.modules.terraform.deployment.resource.TfStateResourceInstance;
+import org.springframework.stereotype.Component;
+
+/**
+ * Terraform resource handler for Huawei.
+ */
+@Component
+@Slf4j
+public class HuaweiTerraformResourceHandler implements DeployResourceHandler {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * Handler of HuaweiCloud for the DeployResult.
+     *
+     * @param deployResult the result of the deployment.
+     */
+    @Override
+    public void handler(DeployResult deployResult) {
+        List<DeployResourceEntity> xpResourceList = new ArrayList<>();
+        TfState tfState;
+        try {
+            var stateFile = deployResult.getRawResources().get("stateFile");
+            if (!stateFile.getClass().equals(String.class)) {
+                throw new RuntimeException("stateFile is unsupported.");
+            }
+            tfState = objectMapper.readValue(stateFile.toString(), TfState.class);
+        } catch (IOException ex) {
+            log.error("Parse terraform state content failed.");
+            throw new TerraformExecutorException("Parse terraform state content failed.", ex);
+        }
+        for (TfStateResource tfStateResource : tfState.getResources()) {
+            DeployResourceEntity deployResourceEntity = new DeployResourceEntity();
```

Comment:
If i understood it right, we are reading the result of the TF execution and then writing data entries. Should it be the other way round? that we write DB entities first when the request is received and later only update the status of it based on the result?